sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -74,31 +79,56 @@ public static SclRootAdapter updateHeader(@NonNull SCL scd, @NonNull HeaderDTO h
         return sclRootAdapter;
     }
 
-
     public static IEDAdapter addIED(SCL scd, String iedName, SCL icd) throws ScdException {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        return sclRootAdapter.addIED(icd,iedName);
+        return sclRootAdapter.addIED(icd, iedName);
     }
 
-    public static Optional<CommunicationAdapter> addSubnetworks(SCL scd, Set<SubNetworkDTO> subNetworks) throws ScdException {
+    public static Optional<CommunicationAdapter> addSubnetworks(SCL scd, Set<SubNetworkDTO> subNetworks, Optional<SCL> icd) throws ScdException {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        CommunicationAdapter communicationAdapter = null;
-        if(!subNetworks.isEmpty()) {
+        CommunicationAdapter communicationAdapter;
+        if (!subNetworks.isEmpty()) {
             communicationAdapter = sclRootAdapter.getCommunicationAdapter(true);
 
             for (SubNetworkDTO subNetworkDTO : subNetworks) {
                 String snName = subNetworkDTO.getName();
                 String snType = subNetworkDTO.getType();
                 for (ConnectedApDTO accessPoint : subNetworkDTO.getConnectedAPs()) {
-                    communicationAdapter.addSubnetwork(snName, snType,
-                            accessPoint.getIedName(), accessPoint.getApName());
+                    String iedName = accessPoint.getIedName();
+                    String apName = accessPoint.getApName();
+                    communicationAdapter.addSubnetwork(snName, snType, iedName, apName);
+
+                    Optional<SubNetworkAdapter> subNetworkAdapter = communicationAdapter.getSubnetworkByName(snName);
+                    if (subNetworkAdapter.isPresent()) {
+                        ConnectedAPAdapter connectedAPAdapter = subNetworkAdapter.get()
+                                .getConnectedAPAdapter(iedName, apName);
+                        copyAddressAndPhysConnFromIcd(apName, icd, connectedAPAdapter);
```

Comment:
```suggestion
                         connectedAPAdapter.copyAddressAndPhysConnFromIcd(icd);
```

Reply:
You're right, I have commited the modification